solar_classes_methods.rb

```diff
@@ -0,0 +1,179 @@
+class Planet
+  attr_reader :name
+  attr_accessor :year_length, :distance_from_sun, :diameter
+
+  def initialize(planet_name)
+    @name = planet_name
+  end
+
+  def planet_summary
+    return "#{name}: \n Takes #{year_length} Earth years to orbit the sun. \n It's distance from the sun is #{distance_from_sun} km. \n #{name}\'s diameter is #{diameter} km."
+  end
+end
+
+class SolarSystem
+  attr_reader :planets, :sys_name # evaluate if this was a good choice for planets
+  # attr_accessor :planets
+
+  def initialize(planets, sys_name)
+    @planets = planets
+    @sys_name = sys_name
+    @all_summary = Array.new
+    @planet_names = Array.new
+    @planet_hash = Hash.new
+    create_all_summary
+    create_planet_hash
+    planet_names
+
+  end
+
+  def add_new_planet(new_planet)
+    @planets << new_planet
+    create_planet_hash
+    create_all_summary
+  end
+
+  def create_planet_hash # this method ties the index +1 of the planet to the instance of the planet, for selection in the menu
+    # @planet_hash = Hash[@planet_names.zip(@planets)]
+    @planets.each.with_index(1) do |planet, index|
+      @planet_hash[index] = planet
+    end
+  end
+
+  def planet_hash
+    @planet_hash
+  end
+
+  def planet_names # creates an array of planet_names maybe this should just be built in to create_planet_hash - same info can be accessed from there.
+    @planets.each do |planet|
+      @planet_names << planet.name
+    end
+  end
+
+  def listplanets
+    numbered_planets = ["0: List all summaries"]
+    @planets.each.with_index(1) do |planet, index|
+      numbered_planet = "#{index}: #{planet.name}"
+      numbered_planets << numbered_planet
```

Comment:
Seems like you could use the `@planet_names` array here to simplify this code a bit.